src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1546,6 +1608,49 @@ CreateHashDistributedTableShards(Oid relationId, int shardCount,
 }
 
 
+/*
+ * CreateHashDistributedTableShards creates the shard of given null-shard-key
+ * distributed table.
+ */
+static void
+CreateNullShardKeyDistTableShard(Oid relationId, Oid colocatedTableId,
+								 bool localTableEmpty)
+{
+	/*
+	 * Decide whether to use exclusive connections per placement or not. Note that
+	 * if the local table is not empty, we cannot use sequential mode since the COPY
+	 * operation that'd load the data into shards currently requires exclusive
+	 * connections.
+	 */
+	bool useExclusiveConnection = false;
+	if (RegularTable(relationId))
+	{
+		useExclusiveConnection = CanUseExclusiveConnections(relationId,
+															localTableEmpty);
+	}
+
+	if (colocatedTableId != InvalidOid)
+	{
+		/*
+		 * We currently allow concurrent distribution of colocated tables (which
+		 * we probably should not be allowing because of foreign keys /
+		 * partitioning etc).
+		 *
+		 * We also prevent concurrent shard moves / copy / splits) while creating
+		 * a colocated table.
+		 */
+		AcquirePlacementColocationLock(colocatedTableId, ShareLock,
+									   "colocate distributed table");
+
+		CreateColocatedShards(relationId, colocatedTableId, useExclusiveConnection);
+	}
+	else
+	{
+		CreateNullKeyShard(relationId, useExclusiveConnection);
```

Comment:
wonder if we should add something like RoundRobin to the name to indiciate that this function is not to be used for co-located null key table creation